Sources/Xcodeproj/generate().swift

```diff
@@ -148,28 +148,30 @@ func findDirectoryReferences(path: AbsolutePath) throws -> [AbsolutePath] {
         if $0.suffix == ".xcodeproj" { return false }
         if $0.suffix == ".playground" { return false }
         if $0.basename.hasPrefix(".") { return false }
-        if isReservedDirectory($0.basename) { return false }
+        if isReservedDirectory($0) { return false }
         return isDirectory($0)
     }
 }
 
-func isReservedDirectory(_ directory: String) -> Bool {
-    return isPackageDirectory(directory) || isSourceDirectory(directory) || isTestDirectory(directory)
+func isReservedDirectory(_ path: AbsolutePath) -> Bool {
```

Comment:
I actually think this was better as a String given that it is just the name of an individual path component.